docs/user/unsupported/mbstyle.rst

```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
 Internally the extension provides greater access to the parsed style:
 
 .. code-block:: java
-  
+
     MBStyle mbstyle = MapBoxStyleParser.parse( reader );
 
     // pull back all layers for a provided source
     List<MBLayer> layers = mbstyle.layers( "mapbox://mapbox.mapbox-streets-v6" );
 
     // pull back selected layers for a provided source
-    List<MBLayer> layers = mbstyle.layers( "mapbox://mapbox.mapbox-streets-v6", "water" ); 
+    List<MBLayer> layers = mbstyle.layers( "mapbox://mapbox.mapbox-streets-v6", "water" );
 
     //Which can be used to Generate Styler Layer Descriptor:
-    StyleLayerDescriptor sld = MBStyleTransformer.transform( layers );
\ No newline at end of file
+    StyleLayerDescriptor sld = MBStyleTransformer.transform( layers );
+
+MapBox Types
+^^^^^^^^^^^^
+ copied from the `MapBox Style Specification <https://www.mapbox.com/mapbox-gl-js/style-spec/>`_
+  Color
+  '''''
+    Colors are written as JSON strings in a variety of permitted formats: HTML-style hex values, rgb, rgba, hsl, and hsla. Predefined HTML colors names, like yellow and blue, are also permitted.
+
+      .. code-block:: json
+
+        {
+          "line-color": "#ff0",
+          "line-color": "#ffff00",
+          "line-color": "rgb(255, 255, 0)",
+          "line-color": "rgba(255, 255, 0, 1)",
+          "line-color": "hsl(100, 50%, 50%)",
+          "line-color": "hsla(100, 50%, 50%, 1)",
+          "line-color": "yellow"
+        }
+    Especially of note is the support for hsl, which can be easier to reason about than rgb().
+
+  Enum
+  ''''
+    One of a fixed list of string values. Use quotes around values.
+
+    .. code-block:: json
+
+      {
+        "text-transform": "uppercase"
+      }
+
+  String
+  ''''''
+    A string is basically just text. In Mapbox styles, you're going to put it in quotes. Strings can be anything, though pay attention to the case of text-field - it actually will refer to features, which you refer to by putting them in curly braces, as seen in the example below.
+
+    .. code-block:: json
+
+      {
+        "text-field": "{MY_FIELD}"
```

Comment:
We list a string, but we show a field, need two examples here.

Note: Why the difference? You can also have an identity function to generate an enum from a field so we need to keep these examples easy to follow.